Feat/on 84 create staff endpoint

tongzxc opened the pull request.

Commit:
feat: create /v1/staffs endpoint and to retrieve all staffs

**RELATES TO**: ON-84

Commit:
feat: create query param for staff status (active and disabled)

**RELATES TO**: ON-84

Commit:
merge develop to ON-84

Commit:
add Beneficiary, Refeee and Case in models

Commit:
pull develop to ON-84

Commit:
Merge branch 'develop' of https://github.com/TechLadies/onesg-app into feat/on-84-create-staff-endpoint

farablanco commented on src/controllers/routers/staffs.js:
```diff
@@ -0,0 +1,40 @@
+'use strict';
+
+const url = require('url');
+const querystring = require('querystring');
```
express has a method to check query parameters, we don't need another package to do so
pls remove the package from package.json (if it was saved) and remove this require here.

tongzxc replied:
changed to use req.query

farablanco commented on src/controllers/routers/staffs.js:
```diff
@@ -0,0 +1,40 @@
+  // to parse the full url to retrieve all query params
+  const parsedUrl = url.parse(fullUrl);
+  // to break down into individual query params
+  const parsedQueries = querystring.parse(parsedUrl.query);
```
use express' method

tongzxc replied:
changed to use req.query

farablanco commented on src/controllers/routers/staffs.js:
```diff
@@ -0,0 +1,40 @@
+    const results = await Staff.query().where('status', 'DISABLED');
+    return res.status(200).json({ results });
+  }
+  const results = await Staff.query();
```
what happens here is if "active" or "disabled" are not matched, then all staff record are returned
since we don't ensure that status is limited to active and disabled
I think we should have that check

tongzxc replied:
included a check to check if !== disabled or !== active, throw an error 
return next(new InvalidInput(`Status can only be 'active' or 'disabled', ${parsedQueries.status} is invalid, `));

Commit:
Merge branch 'develop' of https://github.com/TechLadies/onesg-app into feat/on-84-create-staff-endpoint

Commit:
feat: remove url and querystring and update status validation

if status is not active or disabled, throw error
use req.query

**RELATES TO**: ON-84

Commit:
Merge branch 'develop' of https://github.com/TechLadies/onesg-app into feat/on-84-create-staff-endpoint

farablanco commented on src/controllers/routers/index.js:
```diff
@@ -9,6 +9,9 @@
 const cases = require('./cases');
 const referees = require('./referees');
+const staffs = require('./staffs');
+// const cases = require('./cases');
```
uncomment

Commit:
remove commented cases from routers > index.js

**RELATES TO**: ON-84

farablanco reviewed (approved):
looks good now
